spec.bs

```diff
@@ -1162,20 +1158,28 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
 1. Let |bidGenerators| be the result of running [=build bid generators map=] with |auctionConfig|.
 1. Let |leadingBidInfo| be a new [=leading bid info=].
 1. Let |queue| be the result of [=starting a new parallel queue=].
+1. Let |capturedAuctionHeaders| be [=this=]'s [=relevant global object=]'s [=associated Document's=]

```

Comment:
There is no `[=this=]` here because this algorithm is not a platform object: https://webidl.spec.whatwg.org/#this. You'll have to access the global/document another way — probably through _global_ if that corresponds to the right global object you want.

Also I wonder how we handle race conditions here. See https://github.com/WICG/turtledove/pull/771#discussion_r1330670633. How does this look in the implementation? Does all of this live in the browser process? Is it possible for a fetch with `adAuctionHeaders: true` to be in-flight at this moment, and alter this map in parallel?

Reply:
Changed to use |global|.

About race condition, Caleb replied under https://github.com/WICG/turtledove/pull/771#discussion_r1330670633. I also have a question about this, so left a comment there.

Reply:
Yep, I responded -- the fetch() is called by ad tech scripts on the page, so the ad tech code needs to await the fetch() before resolving the directFromSellerSignalHeaderAdSlot promise.

Reply:
I guess then it might be good for me to add a note for it, like "needs to await the fetch() before resolving the directFromSellerSignalHeaderAdSlot promise". WDYT, @domfarolino

Reply:
oh just noticed that it seems our implementation uses [=auction config/expects direct from seller signals header ad slot=] to wait for fetch()'s result. Is that right? @caraitto

Reply:
Nope -- it is only to wait for the directFromSellerSignalsHeaderAdSlot promise. It doesn't otherwise wait for the fetch() in any way.

Reply:
Added a [Note:](https://github.com/WICG/turtledove/pull/774/files#diff-6f5a1d8263b0b0c42e2716ba5750e3652e359532647ac934c1c70086ae3ceddaR944) for it. Let me know if there's any suggestion on the wording.